### PR TITLE
Julia 1.0 update

### PR DESCRIPTION
**Pull request by nantonel**
Updating to Julia 1.0

@lostella I think we could take this occasion to register the package, what do you think?

### CODE REVIEW

File: src/calculus/precomposeNonlinear.jl [hunk 1]
```diff
@@ -27,9 +27,9 @@ function (f::PrecomposeNonlinear)(x)
 end
 
 function gradient!(y::D, f::PrecomposeNonlinear{P,T,D,C}, x::D) where {P,T,D,C}
-    A_mul_B!(f.bufC, f.G, x)
+    AbstractOperators.mul!(f.bufC, f.G, x)
     v = gradient!(f.bufC2, f.g, f.bufC)
     J = Jacobian(f.G, x)
-    y = Ac_mul_B!(y, J, f.bufC2)
+    y = AbstractOperators.mul!(y, J', f.bufC2)
```
Comment (lostella) [inline src/calculus/precomposeNonlinear.jl]: Why do we need to fully qualify these calls?

File: src/solvers/solvers_options.jl [hunk 2]
```diff
@@ -22,7 +22,7 @@ Creates an object `PG` containing the options of the Proximal Gradient solvers:
 
 """
 struct PG <: ForwardBackwardSolver
-    kwargs::Array
+    kwargs::Iterators.Pairs
```
Comment (lostella) [inline src/solvers/solvers_options.jl]: I guess this is how kwargs are splatted now... but it used to be an Array of pairs?? Crazy

File: src/calculus/precomposeNonlinear.jl (hunk 1 above)
Comment (nantonel) [reply src/calculus/precomposeNonlinear.jl]: We don't... Forgot to remove that!

File: src/solvers/solvers_options.jl (hunk 2 above)
Comment (nantonel) [reply src/solvers/solvers_options.jl]: apparently yes! :smile:

**Pull request merged**